libs/core/threading_base/include/hpx/threading_base/register_thread.hpp

```diff
@@ -58,6 +58,12 @@ namespace hpx { namespace threads {
 
                 // run and free all registered exit functions for this thread
                 auto* p = get_self_id_data();
+
+                // reset thread descriptions to avoid dangling pointers
+                util::thread_description desc;
+                p->set_description(desc);
+                p->set_lco_description(desc);
+
```

Comment:
My only question with this: is the description now just not going to be printed when a thread is destroyed (this should happen before thread destruction as far as I can tell)? With `annotate(d)_function` we assume that c-strings are static strings (without a way to enforce it of course) and for `std::string` we add them to a map to keep them alive. Would the same mechanism be appropriate here?

Reply:
Where is that map? We can certainly do this. My rationale was that maybe having the thread name available during destruction (which usually is long after the thread has exited) might not be necessary. But I'd be happy to change it to storing `std::string`s in a map, somewhere.

Reply:
It's declared and used a couple of times here: https://github.com/STEllAR-GROUP/hpx/blob/master/libs/core/threading_base/include/hpx/threading_base/annotated_function.hpp, and defined here: https://github.com/STEllAR-GROUP/hpx/blob/master/libs/core/threading_base/src/annotated_function.cpp.

Reply:
Thanks. I have added support for `std::string` for executor annotations. This however only adds it for `with_annotation`, not for `get_annotation`. However this shouldn't be an issue.

Reply:
Thanks! Yes, getting them as c-strings should be fine for the same reason it should be fine to use them indefinitely as annotations.